src/it/scala/com/databricks/spark/redshift/RedshiftIntegrationSuite.scala

```diff
@@ -0,0 +1,335 @@
+         |   testlong bigint,
+         |   testshort smallint,
+         |   teststring string,
+         |   testtimestamp timestamp
+         | )
+         | using com.databricks.spark.redshift
+         | options(
+         |   url \"$jdbcUrl\",
+         |   tempdir \"$tempDir\",
+         |   dbtable \"$test_table3\"
+         | )
+       """.stripMargin
+    ).collect()
+  }
+
+  test("DefaultSource can load Redshift UNLOAD output to a DataFrame") {
+    QueryTest.checkAnswer(
+      sqlContext.sql("select * from test_table order by testbyte, testbool"),
+      TestUtils.expectedData)
+  }
+
+  test("DefaultSource supports simple column filtering") {
+    QueryTest.checkAnswer(
+      sqlContext.sql("select testbyte, testbool from test_table order by testbyte, testbool"),
+      Seq(
+        Row(null, null),
+        Row(0.toByte, null),
+        Row(0.toByte, false),
+        Row(1.toByte, false),
+        Row(1.toByte, true)))
+  }
+
+  test("DefaultSource supports user schema, pruned and filtered scans") {
+    // scalastyle:off
+    QueryTest.checkAnswer(
+      sqlContext.sql(
+        """
+          |select testbyte, testbool
+          |from test_table
+          |where testbool = true
+          | and teststring = "Unicode's樂趣"
+          | and testdouble = 1234152.12312498
+          | and testfloat = 1.0
+          | and testint = 42
+        """.stripMargin),
+      Seq(Row(1, true)))
+    // scalastyle:on
+  }
+
+  test("DefaultSource using 'query' supports user schema, pruned and filtered scans") {
+    // scalastyle:off
+    QueryTest.checkAnswer(
+      sqlContext.sql(
+        """
+          |select testbyte, testbool
+          |from test_table
+          |where testbool = true
+          | and teststring = "Unicode's樂趣"
+          | and testdouble = 1234152.12312498
+          | and testfloat = 1.0
+          | and testint = 42
+        """.stripMargin),
+      Seq(Row(1, true)))
+    // scalastyle:on
+  }
+
+  // TODO: test overwrite; test overwite that fails.
+
+  test("Append SaveMode doesn't destroy existing data") {
+    val extraData = Seq(
+      Row(2.toByte, false, null, -1234152.12312498, 100000.0f, null, 1239012341823719L,
+        24.toShort, "___|_123", null))
+
+    sqlContext.createDataFrame(sc.parallelize(extraData), TestUtils.testSchema).write
+      .format("com.databricks.spark.redshift")
+      .option("url", jdbcUrl)
+      .option("dbtable", test_table3)
+      .option("tempdir", tempDir)
+      .mode(SaveMode.Append)
+      .saveAsTable(test_table3)
+
+    QueryTest.checkAnswer(
+      sqlContext.sql("select * from test_table3 order by testbyte, testbool"),
+      TestUtils.expectedData ++ extraData)
+  }
+
+  test("Respect SaveMode.ErrorIfExists when table exists") {
+    val rdd = sc.parallelize(TestUtils.expectedData.toSeq)
+    val df = sqlContext.createDataFrame(rdd, TestUtils.testSchema)
+
+    // Check that SaveMode.ErrorIfExists throws an exception
+    intercept[Exception] {
```

Comment:
Carryover from the other tests that these were inspired by.  I'll fix in both suites.